baybe/searchspace/discrete.py

```diff
@@ -671,6 +679,29 @@ def transform(
         return comp_rep
 
 
+def _apply_pandas_constraint_filter(
```

Comment:
The inplace/return logic is not very consistent in both methods. Both versions currently operate inplace, but the pandas version returns `None` (btw: return type hint missing) while the polars version still returns the modified dataframe. I think this should be handled identically in both methods. Perhaps we should just always return the dataframe (no matter if modified inplace or not) and also add a boolean `inplace` argument to control it?

Reply:
@Scienfitz

Reply:
Yes, I think @Scienfitz can take over from here

Reply:
@rjavadi can the polars filtering even be done `inplace`? If not this request is obsolete (unless you want to add unnecessary copying to the pandas version @AdrianSosic )

Reply:
Can't answer the first question since I don't know, but I think it makes sense to at least have the same output, regardless of whether we (can) do inplace or not.

Reply:
@Scienfitz Unfortunately, the `filter` function doesn't provide an option for in-place filtering.

Reply:
conceptually i guess it doesnt make sense to taLk about inplace filtering for a frame thats not even fully in nemory like a lazyframe,c orrespondingly the pandas utilities would have to oerpate on df copies and return that copy to be the same as polars filter, but I'd find that bad

Reply:
Yes, that's true. I'll leave it up to you then to decide if we can nevertheless return the inplace-modified dataframe. But personally, I'm not a big fan of these types of non-return functions that actually modify something that you want to use in the next step, because they don't allow you to chain operations. For example, every time I use our `add_fake_results` (which works in the same way), I run into the error that my next calling function tells me that `None` doesn't have certain attributes – because I forwarded that empty output to the next processing layer. And then it always requires me to create unnecessary temporary variables. For example
```python
# does not work when `None` is returned
next_operation(add_fake_results(pd.Dataframe(...)))

# current necessary workaround
tmp = pd.DataFrame(...)
add_fake_results(tmp)
next_operation(tmp)
```

I have to say that I quite like `pytorch`s approach here, which cleanly distinguishes in-place vs copy operation in terms of the trailing underscore (e.g. `add` vs `add_`) but the in-place version STILL returns the object. So you don't need to modify your code when switching from one to the other

Reply:
ok then returning simply the object even if altered inplace solves your request? i dont have any reason why it needs noreturn so we can easily alter that (or give it a flag anyway but always return it)

Reply:
I'll put this on the list to do in the integration PR next up, unresolve if the proposed change is not enough